packages/world/src/systems/CraftSystem.sol

```diff
@@ -32,12 +36,18 @@ contract CraftSystem is System {
       caller.requireConnected(station);
     }
 
-    uint128 callerEnergy = transferEnergyToPool(caller, CRAFT_ENERGY_COST);
+    uint256 craftMul = PlayerSkillUtils.getCraftEnergyMultiplierWad(caller, recipe.stationTypeId);
+    uint128 effectiveCost = uint128(FixedPointMathLib.mulWad(CRAFT_ENERGY_COST, craftMul));
+    uint128 callerEnergy = transferEnergyToPool(caller, effectiveCost);
     require(callerEnergy > 0, "Not enough energy to craft");
 
     CraftLib._consumeRecipeInputs(caller, recipe, inputs);
     CraftLib._createRecipeOutputs(caller, recipe);
-    CraftLib._handleEnergyConservation(caller, recipe);
+    uint128 massEnergy = CraftLib._handleEnergyConservation(caller, recipe);
+
+    // Track crafting activity
+    // TODO: this is the output's mass+energy, should we track input instead?
```

Comment:
yea can we combine input and output?

Reply:
hmm but this would be like 2*input if energy is conserved no? so I think it is either input or output but not both. The main difference is for cases like furnaces and dyes, where input >> output

Reply:
ahh right yea its conserved so dont need it